internal/resolver/memory.go

```diff
@@ -48,6 +48,11 @@ func (m *Memory) Tag(_ context.Context, desc ocispec.Descriptor, reference strin
 	return nil
 }
 
+// Delete removes a reference from index map.
+func (m *Memory) Delete(reference string) {
```

Comment:
Just a thoughts: should this be called `Delete` or `UnTag`? 🤔

Reply:
+1 on `Untag`.

Reply:
renamed it as `Untag`.